stun/client.go

```diff
@@ -93,3 +94,27 @@ func (c *Client) Discover() (NATType, *Host, error) {
 	}
 	return discover(conn, serverUDPAddr, c.softwareName, c.logger)
 }
+
+// Keepalive sends and receives a bind request, which ensures the mapping stays open
+// Only applicable when client was created with a connection.
+func (c *Client) Keepalive() (*Host, error) {
+	if c.conn == nil {
+		return nil, errors.New("no connection available")
+	}
+	if c.serverAddr == "" {
+		c.SetServerAddr(DefaultServerAddr)
+	}
+	serverUDPAddr, err := net.ResolveUDPAddr("udp", c.serverAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	_, packet, _, _, host, err := test1(c.conn, serverUDPAddr, c.softwareName)
```

Comment:
I'd have to refactor getting the address to be part of the packet potentially, just to avoid duplication.